tests/components/aquacell/test_config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Test the Aquacell config flow."""
+
+import string
+from unittest.mock import AsyncMock, patch
+
+from aioaquacell import ApiException, AuthenticationFailed
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.aquacell.const import DOMAIN
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.components.aquacell import TEST_RESULT_DATA, TEST_USER_INPUT
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def __mock_authenticate(self, user_name, password) -> string:
+    return "refresh-token"
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.aquacell.config_flow.AquacellApi.authenticate",
+        __mock_authenticate,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            TEST_USER_INPUT,
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["title"] == "test@test.com"
+    assert result2["data"] == {**TEST_RESULT_DATA}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.aquacell.config_flow.AquacellApi.authenticate",
+        side_effect=AuthenticationFailed,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_EMAIL: "test@test.com",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
```

Comment:
1. You can use parametrize to combine these tests and make the exception and the error string a test parameter
2. Both tests should end in `CREATE_ENTRY` to make sure the entry can succeed after hitting an error